akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

```diff
@@ -272,6 +272,20 @@ class SourceSpec extends AkkaSpec with DefaultTimeout {
     }
   }
 
+  "Cycle Source" must {
+
+    "continuously generate the same sequence" in {
+      val expected = Seq(1, 2, 3, 1, 2, 3, 1, 2, 3)
+      Source.cycle(1, 2, 3).grouped(9).runWith(Sink.head).futureValue should ===(expected)
+    }
+
+    "reject creation from empty sequence" in {
+      intercept[IllegalArgumentException] {
+        Source.cycle()
```

Comment:
To my best understanding you are suggesting to change the java dsl implementation to create a new iterator out of array (varargs) and use not `cycle` from scaladsl but `fromIterator`? 
I would also stick to the varargs version of method in javadsl.

Regarding the option of having `def cycle[T](first: T, rest: T*): Source[T, NotUsed] = ???`: I do not think it looks good from client perspective and I would rather stick to the `def cycle[T](elements: T*): Source[T, NotUsed] = ???`